spec/01-unit/01-db/04-dao_spec.lua

```diff
@@ -618,6 +619,55 @@ describe("DAO", function()
       assert.equal(62, entries[12].entity.g)
       assert.equal(63, entries[13].entity.g)
     end)
+
+    it("should call post-delete hook once after concurrent delete", function()
+      finally(function()
+        hooks.clear_hooks()
+      end)
+
+      local hooks_tests = ngx.shared.kong_dao_hooks_tests
+      local post_hook = spy.new(function() end)
+
+      hooks.register_hook("dao:delete:post", function()
+        post_hook()
+      end)
+
+      local schema = Schema.new({
+        name = "Baz",
+        primary_key = { "id" },
+        fields = {
+          { id = { type = "number" } },
+        }
+      })
+
+      local strategy = {
+        select = function()
+          return { id = 1 }
+        end,
+        delete = function(pk, _)
+          local counter = hooks_tests:incr("delete_counter", 1, 0)
+          if counter == 1 then
+            return true
+          end
+          return nil
+        end
+      }
+
+      local dao = DAO.new({}, schema, strategy, errors)
+
+      local co = {}
+      local co_count = 10
+      for i = 1, co_count do
+        co[i] = coroutine.create(function()
```

Comment:
We don't even need to run it in a coroutine, just a straight call to `dao:delete({ id = 1 })` twice is sufficient (this fix does not need concurrency to be tested).